### PR TITLE
add Murphy diagrams to compare 1 or multiple models

### PR DESCRIPTION
**Pull request by mayer79**
Implements simple Murphy diagrams as  queries in https://github.com/mayer79/MetricsWeighted/issues/15

### CODE REVIEW

File: R/murphy_diagram.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#' y <- 1:10
+#' predicted <- 1.1 * y
+#' murphy_diagram(y, predicted, theta = seq(0.9, 1.2, by = 0.01))
+#' two_models <- cbind(m1 = predicted, m2 = 1.2 * y)
+#' murphy_diagram(y, two_models, theta = seq(0.9, 1.3, by = 0.01))
+#' @seealso \code{\link{elementary_score}}.
+murphy_diagram <- function(actual, predicted, w = NULL,
+                           alpha = 0.5,
+                           theta = seq(-2, 2, length.out = 100),
+                           type = c("expectile", "quantile"),
```
Comment (lorentzenchr) [inline R/murphy_diagram.R]: Several questions here:

1. Is `type` a good name?
2. Is `c("expectile", "quantile")` a good default or would just `"expectiles"` be more suitable.

File: R/murphy_diagram.R [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#' Murphy diagram
+#'
+#' Murphy diagram for elementary scores. Can be used to compare performance of one or multiple models across ranges of thetas for elementary scores. If the plot needs to be customized, set \code{plot = FALSE} to get the resulting data instead of the plot.
+#'
+#' @importFrom graphics matplot legend
+#' @param actual Observed values.
+#' @param predicted Predicted values.
+#' @param w Optional case weights.
+#' @param alpha Level of expectile resp. quantile.
```
Comment (lorentzenchr) [inline R/murphy_diagram.R]: Maybe mention again that the default `alpha=0.5` corresponds to the expectation resp. median.

File: R/murphy_diagram.R [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#' @param theta Vector of theta values.
+#' @param type One of "expectile", "quantile".
```
Comment (lorentzenchr) [inline R/murphy_diagram.R]: The default is a vector, so vetors are allowed, too?

File: vignettes/MetricsWeighted.Rmd [hunk 4]
```diff
@@ -248,5 +250,17 @@ perf$`Tweedie p` <- as.numeric(as.character(perf$`Tweedie p`))
 plot(`R-squared` ~ `Tweedie p`, data = perf, type = "s")
 ```
 
+### Murphy diagrams
+
+The same logic as in the last example can be used to create so-called *Murphy diagrams* [@gneiting]. The function `murphy_diagram()` wraps above calls and allows to get elementary scores for one or multiple models across a range of theta values.
```
Comment (lorentzenchr) [inline vignettes/MetricsWeighted.Rmd]: Just a thought: We could be nice and also link to https://cran.r-project.org/package=murphydiagram.

**Review by lorentzenchr — APPROVED** [review]
LGTM. Some (optional) comments/questions.
I'm looking forward to having those plots in the next release:smirk:

File: R/murphy_diagram.R (hunk 3 above)
Comment (mayer79) [reply R/murphy_diagram.R]: `type` has "expectile" as single default, even if it seems to be a vector. This is an R thing to show all possible input values. Other values raise an error in `match.arg()`. Is "expectiles" better than "expectile"?

Comment (lorentzenchr) [reply R/murphy_diagram.R]: :bulb:
"expectile**s**" was a misspelling of mine.

File: R/murphy_diagram.R (hunk 1 above)
Comment (mayer79) [reply R/murphy_diagram.R]: Renamed `type` to `functional`

File: R/murphy_diagram.R (hunk 2 above)
Comment (mayer79) [reply R/murphy_diagram.R]: Added, thx.

**Comment by lorentzenchr** [comment]
🎉 :rocket:

**Pull request merged**